syntax = "proto3";

service AuthService {
    rpc RegisterAccount (RegisterRequest) returns (RegisterResponse);
    rpc LoginAccount (LoginRequest) returns (LoginResponse);
}

message RegisterRequest {
    string username = 1;
    string password = 2;
    double initial_amount = 3;
    string bank_name = 4;
}

message RegisterResponse {
    string account_number = 1;
    string message = 2;
    string bank_name=3;
    bool success=4;
}

message LoginRequest {
    string username = 1;
    string password = 2;
    string bank_name = 3;
}

message LoginResponse {
    string account_number = 1;
    string key = 2;
    string message = 3;
}